lib/services/access_request_test.go

```diff
@@ -1415,3 +1442,152 @@ func TestPruneRequestRoles(t *testing.T) {
 		})
 	}
 }
+
+// TestPendingTTL verifies that the TTL for the Access Request gets reduced by
+// requested access time and lifetime of the requesting certificate.
+func TestPendingTTL(t *testing.T) {
+	clock := clockwork.NewFakeClock()
+	now := clock.Now().UTC()
+
+	tests := []struct {
+		desc          string
+		expiry        time.Time
+		identity      tlsca.Identity
+		maxSessionTTL time.Duration
+		expectedTTL   time.Duration
+	}{
+		{
+			desc:          "access request with ttl",
+			expiry:        now.Add(8 * time.Hour),
+			identity:      tlsca.Identity{Expires: now.Add(10 * time.Hour)},
+			maxSessionTTL: 10 * time.Hour,
+			expectedTTL:   8 * time.Hour,
+		},
+		{
+			desc:          "access request without ttl",
+			expiry:        time.Time{},
+			identity:      tlsca.Identity{Expires: now.Add(10 * time.Hour)},
+			maxSessionTTL: 10 * time.Hour,
+			expectedTTL:   defaults.PendingAccessDuration,
+		},
+		{
+			desc:          "access request truncation by identity expiration",
+			expiry:        time.Time{},
+			identity:      tlsca.Identity{Expires: now.Add(13 * time.Hour)},
+			maxSessionTTL: 13 * time.Minute,
+			expectedTTL:   13 * time.Minute,
```

Comment:
`desc` says this tests truncation by identity expiration, but the identity expires in 13 hours while the role maxSessionTTL is 13 minutes, I think this case is currently functionally identical to the following one

```suggestion
			desc:          "access request truncation by identity expiration",
			expiry:        time.Time{},
			identity:      tlsca.Identity{Expires: now.Add(12 * time.Minute)},
			maxSessionTTL: 13 * time.Minute,
			expectedTTL:   12 * time.Minute,
```